src/test/scala/org/neo4j/spark/DataSourceReaderNeo4j41xTSE.scala

```diff
@@ -0,0 +1,92 @@
+package org.neo4j.spark
+
+import org.apache.spark.sql.DataFrame
+import org.junit.Assert.assertEquals
+import org.junit.{Assume, BeforeClass, Test}
+import org.neo4j.driver.summary.ResultSummary
+import org.neo4j.driver.{SessionConfig, Transaction, TransactionWork}
+
+object DataSourceReaderNeo4j41xTSE {
+  @BeforeClass
+  def checkNeo4jVersion() {
+    Assume.assumeTrue(TestUtil.neo4jVersion().startsWith("4.1"))
```

Comment:
I would change this in this way'
```suggestion
    val neo4jVersion = TestUtil.neo4jVersion()
    Assume.assumeTrue(!neo4jVersion.startsWith("3.5") && !neo4jVersion.startsWith("4.0"))
```

in this way should work also with newer releases such as 4.2 and so on, otherwise we need to clone the test for every new Neo4j version.

WDYT?

Reply:
Makes totally sense! On it